modules/ROOT/pages/patterns/reference.adoc

```diff
@@ -0,0 +1,1302 @@
+
+Take this example graph:
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE (filipa: Person {name: 'Filipa'}),
+  (anders: Person {name: 'Anders'}),
+  (dishad: Person {name: 'Dishad'}),
+  (filipa)-[:KNOWS]->(anders),
+  (anders)-[:KNOWS]->(dishad)
+----
+////
+
+image::patterns_equijoin_reference.svg[width="500", role="middle"]
+
+In the following query, the node variables will be bound to the same nodes:
+
+.Query
+[source, cypher]
+----
+MATCH (a {name: 'Filipa'})<-[r*1..2]-(b)
+MATCH (c)<-[r*1..2]-(d)
+RETURN a = c, b = d, size(r)
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| a = c | b = d | size(r)
+
+| TRUE | TRUE | 1
+| TRUE | TRUE | 2
+
+3+d|Rows: 2
+|===
+
+The list of relationships keeps its order.
+This means that in the following query, where the direction of the variable-length relationship in the second `MATCH` is switched, the equijoin will only match once, when there is a single relationship:
+
+.Query
+[source, cypher]
+----
+MATCH (a {name: 'Filipa'})<-[r*1..2]-(b)
+MATCH (c)-[r*1..2]->(d)
+RETURN a = c, b = d, size(r)
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| a = c | b = d | size(r)
+
+| FALSE | FALSE | 1
+
+3+d|Rows: 1
+|===
+
+The variable `r` can be reversed in order like any list, and made to match the switch in relationship pattern direction:
+
+.Query
+[source, cypher]
+----
+MATCH (a {name: 'Filipa'})<-[r*1..2]-(b)
+WITH a, b, reverse(r) AS s
+MATCH (c)-[s*1..2]->(d)
+RETURN a = d, b = c, size(s)
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| a = d | b = c | size()
+
+| TRUE | TRUE | 1
+| TRUE | TRUE | 2
+
+3+d|Rows: 2
+
+|===
+
+Changing the bounds on subsequent `MATCH` statements will mean that only the overlapping lengths of the quantifier bounds will produce results:
+
+.Query
+[source, cypher]
+----
+MATCH (a {name: 'Filipa'})<-[r*1..2]-(b)
+MATCH (c)<-[r*2..3]-(d)
+RETURN a = c, b = d, size(r)
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="3*<m"]
+|===
+
+| a = c | b = d | size(r)
+
+| TRUE | TRUE | 2
+
+3+d|Rows: 1
+
+|===
+
+Because Cypher only allows paths to traverse a relationship once (see xref:patterns/reference.adoc#graph-patterns-rules-relationship-uniqueness[relationship uniqueness]), repeating a variable-length relationship in the same graph pattern will yield no results.
+For example, this `MATCH` query will never return any results:
+
+[source, role=noheader]
+----
+MATCH (x)-[r*1..2]->(y)-[r*1..2]->(z)
+----
+
+Attempting to repeat a variable-length relationship in a single relationship pattern will raise an error.
+For example, this is not valid:
+
+[source, role=noheader]
+----
+MATCH (x)-[r*1..2]->(y)-[r]->(z)
+----
+
+[[variable-length-relationships-examples]]
+=== Examples
+
+The following pattern matches paths starting with nodes labeled `A` and ending with nodes labeled `B`, that traverse between 2 and 3 relationships of type `R`:
```

Comment:
Numbers up to 12 should be written as words, and after as digits. We are working on a style-guide currently where this will be formalised, but for now I will change it in this chapter.